backend-lib/src/main/rust/lib.rs

```diff
@@ -2169,8 +2157,7 @@ pub extern "C" fn Java_cash_z_ecc_android_sdk_internal_jni_RustBackend_requiresS
     let res = catch_unwind(&mut env, |env| {
         let _span = tracing::info_span!("RustBackend.pcztRequiresSaplingProofs").entered();
```

Comment:
This is the correct name, but the mangled function name is wrong on line 2150. How did this ever work? (I assume it didn't, and so we must have inadequate test coverage.)

Reply:
We cannot rename the JNI method, because the arguments between the extern function and the Backend trait method are identical. You'll see I use different names for several such methods, whereas when the Backend trait method has fewer arguments (e.g. due to us inserting a `db_data` path for the JNI) we use the same method name.

Reply:
It might be better to systematically use a different name (e.g. add an `ffi` prefix), rather than sometimes use the same name and sometimes a different one.